api/src/services/__tests__/programsService.ts

```diff
@@ -0,0 +1,372 @@
+        [curriculumId],
+        matchingPrograms
+      );
+      expect(await listProgramsForCurriculum(curriculumId)).toEqual(
+        matchingPrograms
+      );
+    });
+  });
+
+  describe('findProgram', () => {
+    it('should give the details of the specified program', async () => {
+      const programId = 2;
+      const [matchingProgram] = [programsList[1]];
+      mockQuery(
+        'select `id`, `title`, `start_date`, `end_date`, `time_zone`, `curriculum_id` from `programs` where `id` = ?',
+        [programId],
+        [programsList[1]]
+      );
+      expect(await findProgram(programId)).toEqual(matchingProgram);
+    });
+  });
+
+  describe('listCurriculumActivities', () => {
+    it('should list all available activities for the specified curriculum', async () => {
+      const curriculumId = 1;
+      const matchingCurriculumActivities: CurriculumActivity[] = [
+        curriculumActivitiesList[0],
+        curriculumActivitiesList[1],
+        curriculumActivitiesList[2],
+      ];
+      mockQuery(
+        'select `id`, `title`, `description_text`, `curriculum_week`, `curriculum_day`, `start_time`, `end_time`, `duration`, `activity_type_id`, `curriculum_id` from `activities` where `curriculum_id` = ?',
+        [curriculumId],
+        matchingCurriculumActivities
+      );
+      expect(await listCurriculumActivities(curriculumId)).toEqual(
+        matchingCurriculumActivities
+      );
+    });
+  });
+
+  describe('findCurriculumActivity', () => {
+    it('should give the details of the specified program', async () => {
+      const curriculumActivityId = 4;
+
+      const [, , , matchingCurriculumActivity] = curriculumActivitiesList;
+
+      mockQuery(
+        'select `id`, `title`, `description_text`, `curriculum_week`, `curriculum_day`, `start_time`, `end_time`, `duration`, `activity_type_id`, `curriculum_id` from `activities` where `id` = ?',
+        [curriculumActivityId], // binding param // replace above ? with the value // findProgram() get the first program object, so the limit is 1
+        [matchingCurriculumActivity]
+      );
+
+      expect(await findCurriculumActivity(curriculumActivityId)).toEqual(
+        matchingCurriculumActivity
+      );
+    });
+  });
+
+  describe('listProgramActivities', () => {
+    it('should list all available activities for the specified program', async () => {
+      const programId = 2;
+      const curriculumId = 2;
+
+      const [, matchingProgram] = programsList;
+
+      const matchingCurriculumActivities: CurriculumActivity[] = [
+        curriculumActivitiesList[3],
+        curriculumActivitiesList[4],
+        curriculumActivitiesList[5],
+      ];
+
+      const matchingProgramActivities: ProgramActivity[] = [
+        program2ActivitiesList[0],
+        program2ActivitiesList[1],
+        program2ActivitiesList[2],
+      ];
+
+      // findProgram
```

Comment:
Lines 297 - 318 comments should be removed as they do not provide extra value than the variable names themselves.

Reply:
Extraneous comments have been removed.